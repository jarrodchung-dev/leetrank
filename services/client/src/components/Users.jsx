import React from "react";

const Users = (props) => {
  return (
    <div className="container">
      <div className="columns" id="users">
        {props.users.map((user) => {
          return (
            <div className="level" key={user.id}>
              <div className="level-left">
                <div className="level-item">
                  <p>{user.username}</p>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Users;
