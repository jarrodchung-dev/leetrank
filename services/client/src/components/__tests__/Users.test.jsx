import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";

import Users from "../Users.jsx";

const users = [
  {
    id: 1,
    email: "testuser@example.com",
    username: "testuser",
    active: true
  },
  {
    id: 2,
    email: "testuser2@example.com",
    username: "testuser2",
    active: true
  }
];

describe("<Users />", () => {
  it("displays users passed down as props", () => {
    const wrapper = mount(<Users users={users} />);
    const element = wrapper.find("p");
    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe("testuser");
    expect(element.get(1).props.children).toBe("testuser2");
  });

  it("renders a snapshot successfully", () => {
    const tree = renderer.create(<Users users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
