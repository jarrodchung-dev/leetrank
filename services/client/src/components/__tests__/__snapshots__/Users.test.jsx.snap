// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> renders a snapshot successfully 1`] = `
<div
  className="container"
>
  <div
    className="columns"
    id="users"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <p>
            testuser
          </p>
        </div>
      </div>
    </div>
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <p>
            testuser2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
